### PR TITLE
add open struct option

### PR DESCRIPTION
**Pull request by vishalkuo**

### CODE REVIEW

**Comment by 6** [comment]
Why do we want two different outputs for pheme?

**Comment by vishalkuo** [comment]
1) Deserializing a message into a recursive open struct is opinionated: developers may just want to use a vanilla ruby hash
2) If my message is JSON array, recursive open struct would not properly deserialize it unless i wrap it in an outer object. 

eg.

```
[1] pry(main)> x = [{"x": "y"}]
=> [{:x=>"y"}]
[2] pry(main)> RecursiveOpenStruct.new(x)
=> #<RecursiveOpenStruct {:x=>"y"}=nil>
[3] pry(main)> r[0]
TypeError: 0 is not a symbol nor a string
from /Users/vishalkuo/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/recursive-open-struct-1.0.1/lib/recursive_open_struct.rb:42:in `public_send'
```

**Comment by cabello** [comment]
:shipit:

File: lib/pheme/queue_poller.rb [hunk 1]
```diff
@@ -60,7 +60,11 @@ def parse_csv(message_contents)
 
     def parse_json(message_contents)
       parsed_body = JSON.parse(message_contents)
-      RecursiveOpenStruct.new(parsed_body, recurse_over_arrays: true)
+      if parsed_body.instance_of? Array
```
Comment (6) [inline lib/pheme/queue_poller.rb]: slightly simpler: you can use `#is_a?` here instead.

**Pull request merged**